src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj

```diff
@@ -162,21 +164,39 @@
           (integer? compat-version) double
           :true str))
 
+(schema/defn ^:always-validate warn-if-compat-version-setting-not-supported
+  [parsed-compat-version :- schema/Str]
+  (let [const-compat-version (jruby-internal/get-compat-version
+                              parsed-compat-version)
+        instance-config (RubyInstanceConfig.)]
+    (.setCompatVersion instance-config const-compat-version)
+    (when-not (= const-compat-version (.getCompatVersion instance-config))
+      (log/warnf
+       "%s %s %s"
+       (i18n/trs "compat-version is set to `{0}`, which is not a supported version."
+                 parsed-compat-version)
+       (i18n/trs "Version `{0}` will be used instead." Constants/RUBY_VERSION)
+       (i18n/trs "Setting 'compat-version' is deprecated and will be removed in a future release.")))))
```

Comment:
Yeah, I can add something about JRuby 9k to the message and look into adding a test for it.  A non-mocked test would obviously only be able to assert the generation of the warning when the tests are running under JRuby 9k but that seems okay to me.